OpenSim/Region/OptionalModules/Avatar/FlexiGroups/FlexiGroupsModule.cs

```diff
@@ -1034,14 +1034,29 @@ public UUID CreateGroup(IClientAPI remoteClient, string name, string charter, bo
                 remoteClient.SendCreateGroupReply(UUID.Zero, false, "A group with the same name already exists.");
                 return UUID.Zero;
             }
-            
-            UUID groupID = m_groupData.CreateGroup(grID, name, charter, showInList, insigniaID, membershipFee, openEnrollment, allowPublish, maturePublish, remoteClient.AgentId);
 
-            remoteClient.SendCreateGroupReply(groupID, true, "Group created successfullly");
+            IScene scene = (remoteClient == null) ? m_sceneList[0] : remoteClient.Scene;
+            IMoneyModule mm = scene.RequestModuleInterface<IMoneyModule>();
+            if (mm == null)
+            {
+                remoteClient.SendAgentAlertMessage("Server configuration error - cannot create group without money module.", false);
```

Comment:
This introduces a new semantic: previously the absence of a money module still allowed group creation.  I suggest restoring that semantic.

Reply:
While it is true the absence of the money module allowed it previously (on this one operation), that is because this one operation was completely missing all money-related code/checks.  (Sending an IM to a user also succeeds with no money module because of course that operation is currently free...)

As far as I understand things, the money module is not optional, there are many places that would crash or fail if one was not present.  (One example discovered below is that mesh uploads will crash.)

More importantly, this suggestion would make it inconsistent with the application of the money balance checks used on other operations (e.g. in `HandleUDPUploadRequest` and `NewAgentInventoryRequest` which both have code to return an error to the user if there is no money module.  Clearly the intention is to fail an operation that has a fee if there is no money module available.

However, I notice that `NewAgentInventoryRequest` (the mesh upload case) will crash with a null reference exception just before it attempts to report the insufficient funds if there's no money module defined.  That said, crashing before performing the operation still means it won't succeed when there is no money module.

I modeled this change after the other cases, thus it has the same failure result if there's a fee and no money module installed.  I don't think we should be changing that behavior for this specific new check, just because there wasn't any check at all previously.

Reply:
That makes sense then: always helps to see the larger picture.